src/main/kotlin/de/bund/digitalservice/useid/widget/WidgetController.kt

```diff
@@ -57,6 +63,16 @@ class WidgetController(
             "additionalClass" to ""
         )
 
+        val parsedUserAgent = Parser().parse(userAgent)
+        val incompatibleIOSVersion = hasSupportedMajorOSVersion(parsedUserAgent, "iOS", 14)
+        val incompatibleAndroidVersion = hasSupportedMajorOSVersion(parsedUserAgent, "Android", 8)
+
+        if (incompatibleIOSVersion || incompatibleAndroidVersion) {
```

Comment:
Just a matter of taste but this could be extracted to a method like `isOSSupported()` which would keep the handler method cleaner.